File: server/routes/payments.js
const express = require('express');
const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
const auth = require('../middleware/auth');

const router = express.Router();

// Create payment intent for project posting
router.post('/create-payment-intent', auth, async (req, res) => {
  try {
    const { amount } = req.body; // Amount in cents

    const paymentIntent = await stripe.paymentIntents.create({
      amount,
      currency: 'usd',
      metadata: {
        userId: req.user._id.toString(),
        type: 'project_posting'
      }
    });

    res.json({
      clientSecret: paymentIntent.client_secret
    });
  } catch (error) {
    res.status(500).json({ message: 'Payment error', error: error.message });
  }
});

// Process payment for completed project
router.post('/pay-freelancer', auth, async (req, res) => {
  try {
    const { projectId, amount } = req.body;

    // Create transfer to freelancer (simplified - in production you'd need connected accounts)
    const paymentIntent = await stripe.paymentIntents.create({
      amount,
      currency: 'usd',
      metadata: {
        projectId,
        type: 'freelancer_payment'
      }
    });

    res.json({
      clientSecret: paymentIntent.client_secret
    });
  } catch (error) {
    res.status(500).json({ message: 'Payment error', error: error.message });
  }
});

module.exports = router;

File: server-python/routes/payments.py
from flask import Blueprint, request, jsonify
from routes.middleware import token_required
import stripe
import os

payments_bp = Blueprint('payments', __name__)

stripe.api_key = os.getenv('STRIPE_SECRET_KEY')

@payments_bp.route('/create-payment-intent', methods=['POST'])
@token_required
def create_payment_intent(current_user):
    try:
        data = request.json or {}
        amount = data.get('amount')
        
        if not amount:
            return jsonify({"message": "Amount is required"}), 400
            
        payment_intent = stripe.PaymentIntent.create(
            amount=amount,
            currency='usd',
            metadata={
                'userId': str(current_user.id),
                'type': 'project_posting'
            }
        )
        
        return jsonify({
            "clientSecret": payment_intent.client_secret
        }), 200
    except Exception as e:
        return jsonify({"message": "Payment error", "error": str(e)}), 500

@payments_bp.route('/pay-freelancer', methods=['POST'])
@token_required
def pay_freelancer(current_user):
    try:
        data = request.json or {}
        project_id = data.get('projectId')
        amount = data.get('amount')
        
        if not amount:
            return jsonify({"message": "Amount is required"}), 400
            
        payment_intent = stripe.PaymentIntent.create(
            amount=amount,
            currency='usd',
            metadata={
                'projectId': str(project_id),
                'type': 'freelancer_payment'
            }
        )
        
        return jsonify({
            "clientSecret": payment_intent.client_secret
        }), 200
    except Exception as e:
        return jsonify({"message": "Payment error", "error": str(e)}), 500
